fast64_internal/oot/c_writer/oot_scene_bootup.py

```diff
@@ -0,0 +1,139 @@
+        + f"#endif\n"
+    )
+
+    writeFile(os.path.join(exportPath, "include/config/config_bootup.h"), data)
+
+
+def getParamsFromOptions(options: Any) -> tuple[str, str]:
+    timeOfDay = (
+        "NEXT_TIME_DAY"
+        if options.headerOption == "Child Day" or options.headerOption == "Adult Day"
+        else "NEXT_TIME_NIGHT"
+    )
+
+    linkAge = (
+        "LINK_AGE_ADULT"
+        if options.headerOption == "Adult Day" or options.headerOption == "Adult Night"
+        else "LINK_AGE_CHILD"
+    )
+
+    return timeOfDay, linkAge
+
+
+# converts ascii text to format for save file name.
+# see src/code/z_message_PAL.c:Message_Decode()
+def stringToSaveNameBytes(name: str) -> bytearray:
+    specialChar = {
+        " ": 0x3E,
+        ".": 0x40,
+        "-": 0x3F,
+    }
+
+    result = bytearray([0x3E] * 8)
+
+    if len(name) > 8:
+        raise PluginError("Save file name for scene bootup must be 8 characters or less.")
+    for i in range(len(name)):
+        value = ord(name[i])
+        if name[i] in specialChar:
+            result[i] = specialChar[name[i]]
+        elif value >= ord("0") and value <= ord("9"):  # numbers
+            result[i] = value - ord("0")
+        elif value >= ord("A") and value <= ord("Z"):  # uppercase
+            result[i] = value - ord("7")
+        elif value >= ord("a") and value <= ord("z"):  # lowercase
+            result[i] = value - ord("=")
+        else:
+            raise PluginError(
+                name + " has some invalid characters and cannot be used as a save file name for scene bootup."
+            )
+
+    return result
+
+
+class OOT_ClearBootupScene(bpy.types.Operator):
+    bl_idname = "object.oot_clear_bootup_scene"
+    bl_label = "Undo Boot To Scene"
+    bl_options = {"REGISTER", "UNDO", "PRESET"}
+
+    def execute(self, context):
+        clearBootupScene(bpy.path.abspath(context.scene.ootDecompPath))
+        self.report({"INFO"}, "Success!")
+        return {"FINISHED"}
+
+
+ootEnumBootMode = [("Play", "Play", "Play"), ("File Select", "File Select", "File Select")]
+
+
+class OOTBootupSceneOptions(bpy.types.PropertyGroup):
+    bootToScene: bpy.props.BoolProperty(default=False, name="Boot To Scene")
+    overrideHeader: bpy.props.BoolProperty(default=False, name="Override Header")
+    headerOption: bpy.props.EnumProperty(items=ootEnumHeaderMenuComplete, name="Header", default="Child Day")
+    spawnIndex: bpy.props.IntProperty(name="Spawn", min=0)
+    newGameOnly: bpy.props.BoolProperty(default=False, name="Override Scene On New Game Only")
+    newGameName: bpy.props.StringProperty(default="Link", name="New Game Name")
+    bootMode: bpy.props.EnumProperty(default="Play", name="Boot Mode", items=ootEnumBootMode)
+
+    # see src/code/z_play.c:Play_Init() - can't access more than 16 cutscenes?
+    cutsceneIndex: bpy.props.IntProperty(min=4, max=19, default=4, name="Cutscene Index")
```

Comment:
yeah, there's no way to load the 17th cutscene header if a scene has one, due to how `cutsceneIndex` values are (0xFFF0-0xFFFF)
I don't know exactly what's going on with this stuff though
there's also this weird bit in play_init
```c
if (gSaveContext.cutsceneIndex == 0xFFFD) {
    gSaveContext.cutsceneIndex = 0;
}
```
which'd suggest you can't use the 14th cutscene header (index 0xD=13)
:shrug: not a big deal for the purpose of this tool

(and fwiw it'd make more sense that this value starts at 0, like "cutscene 0")

Reply:
Yeah, I thought about 0 indexing, but at this point the first index=4 appears so many times in the UI/codebase that I thought it would be better to keep it consistent.